### PR TITLE
[fix](join) incorrect result of right anti join with nullable

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

If the probe column on the left side is not nullable and the build column on the right is nullable, the anti join's result is incorrect.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

File: be/src/vec/exec/join/vhash_join_node.cpp [hunk 1]
```diff
@@ -1080,6 +1082,15 @@ Status HashJoinNode::_extract_join_column(Block& block, ColumnUInt8::MutablePtr&
                     raw_ptrs[i] = &col_nested;
                 }
             } else {
+                if constexpr (!BuildSide) {
+                    if (_join_op == TJoinOp::RIGHT_ANTI_JOIN &&
```
Comment (BiteTheDDDDt) [inline be/src/vec/exec/join/vhash_join_node.cpp:1086]: Does only right anti join need to be converted?

Comment (mrhhsg) [reply be/src/vec/exec/join/vhash_join_node.cpp:1086]: Yes,  right join and ful join will be converted in `_convert_block_to_null`.

**Review by BiteTheDDDDt — APPROVED** [review]

**Comment by mrhhsg** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by mrhhsg** [comment]
run buildall

**Pull request merged**